packages/paste-core/components/menu/src/MenuItem.tsx

```diff
@@ -1,16 +1,53 @@
 import * as React from 'react';
 import {secureExternalLink} from '@twilio-paste/anchor';
 import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import type {BoxStyleProps} from '@twilio-paste/box';
 import {MenuPrimitiveItem} from '@twilio-paste/menu-primitive';
 
-import type {MenuItemProps} from './types';
-import {MenuItemVariants} from './constants';
+import type {MenuItemProps, MenuItemVariant} from './types';
 import {MenuGroupContext} from './MenuGroup';
+import {MenuItemVariants} from './constants';
+
+const variantStyles: Record<MenuItemVariant, BoxStyleProps> = {
+  [MenuItemVariants.DEFAULT]: {
+    color: 'colorText',
+    _hover: {
+      color: 'colorTextPrimary',
+      backgroundColor: 'colorBackgroundPrimaryWeakest',
+      borderColor: 'colorBorderPrimary',
+    },
+  },
+  [MenuItemVariants.DESTRUCTIVE]: {
+    color: 'colorTextLinkDestructive',
```

Comment:
note: just to summarize 16th min, we have a ticket in the backlog to add `colorTextDestructive` tokens + we'll update menu after that